modin/experimental/core/storage_formats/pandas/small_query_compiler.py

```diff
@@ -0,0 +1,1305 @@
+    is_series : bool, default: False
+        If True, the passed frame will always be squeezed to a series.
+    squeeze_series : bool, default: False
+        If True, the passed frame will always be squeezed to a series if there is a single column named "__reduced__".
+    squeeze_args : bool, default: False
+        If True, all passed arguments will be squeezed.
+    squeeze_kwargs : bool, default: False
+        If True, all passed key word arguments will be squeezed.
+    return_modin : bool, default: True
+        If True, the result will always try to convert to DataFrame or Series.
+    in_place : bool, default: False
+        If True, the specified function will be applied on the passed frame in place.
+    df_copy : bool, default: False
+        If True, the specified function will be applied to a copy of the passed frame.
+    filter_kwargs : list, default: []
+        List of key word argument names to remove.
+
+    Returns
+    -------
+    callable(pandas.DataFrame, *args, **kwargs) -> pandas.DataFrame
+        Function to be applied to the frame.
+    """
+
+    def caller(query_compiler, *args, **kwargs):
+        df = query_compiler._pandas_frame
+        if df_copy:
+            df = df.copy()
+        if is_series:
+            df = df.squeeze(axis=1)
+        exclude_names = [
+            # "broadcast",
+            "fold_axis",
+            # "squeeze_self",
+            # "squeeze_value",
+            "ignore_indices",
+        ] + filter_kwargs
+        kwargs = kwargs.copy()
+        for name in exclude_names:
+            kwargs.pop(name, None)
+        args = try_cast_to_pandas(args, squeeze=squeeze_args)
+        kwargs = try_cast_to_pandas(kwargs, squeeze=squeeze_kwargs)
+        result = func(df, *args, **kwargs)
+        if in_place:
+            result = df
+        if not (return_modin or isinstance(result, (pandas.Series, pandas.DataFrame))):
+            return result
+        if isinstance(result, pandas.Series):
+            if result.name is None:
+                result.name = MODIN_UNNAMED_SERIES_LABEL
+            result = result.to_frame()
+
+        return query_compiler.__constructor__(result)
+
+    return caller
+
+
+@_inherit_docstrings(BaseQueryCompiler)
+class SmallQueryCompiler(BaseQueryCompiler):
+    """
+    Query compiler for the pandas storage format.
+
+    This class translates common query compiler API to default all methods
+    to pandas.
+
+    Parameters
+    ----------
+    pandas_frame : pandas.DataFrame
+        Modin Frame to query with the compiled queries.
+    """
+
+    def __init__(self, pandas_frame):
+        assert InitializeWithSmallQueryCompilers.get()
+        if hasattr(pandas_frame, "_to_pandas"):
+            pandas_frame = pandas_frame._to_pandas()
+        if is_scalar(pandas_frame):
+            pandas_frame = pandas.DataFrame([pandas_frame])
+        elif not isinstance(pandas_frame, pandas.DataFrame):
+            pandas_frame = pandas.DataFrame(pandas_frame)
```

Comment:
Why so many conditions? Don't we always pass a pandas DataFrame in to this constructor?

Reply:
The dataframe can be scalar when calling some pandas functions [here](https://github.com/arunjose696/modin/blob/arun-sqc/modin/experimental/core/storage_formats/pandas/small_query_compiler.py#L605 ) , for eg in df.equals(). Thus we need to address this case in constructor.

The  case  `if hasattr(pandas_frame, "_to_pandas")` was included to allow casting from PandasQC to PlainPandasQueryCompiler which may be needed later.